IsogeomGenerator/generate_isogeom.py

```diff
@@ -0,0 +1,505 @@
+        vizualization:
+
+        generate_isogeom moab -lf tmp/levelfile -t E1 1.0 -t E2 10.0 -v
+    """
+    moab_parser = subparsers.add_parser('moab',
+                                        description=moab_description,
+                                        usage=moab_usage,
+                                        epilog=moab_examples,
+                                        formatter_class=argparse.RawDescriptionHelpFormatter)
+    set_shared_options(moab_parser, moab=True)
+    set_moab_only_options(moab_parser)
+    moab_parser.set_defaults(which='moab')
+
+    args = parser.parse_args()
+    return args
+
+
+def check_level_gen(args):
+    """Check that correct args were supplied for generating levels.
+
+    If the min/max values or N are not supplied, then raise error.
+
+    Input:
+    ------
+        args: set of ArgumentParser args
+    """
+    if (args.extN is None) or (args.N is None):
+        raise RuntimeError("Min/Max level values (-lx) and number of levels " +
+                           "(-N) must be set to use --generatelevels option.")
+
+
+def get_levels(args, g):
+    """Get level information depending on supplied args.
+
+    Input:
+    ------
+        args: set of ArgumentParser args
+        g: IsoVolume instance
+    """
+    # collect level information:
+    if args.levelfile is not None:
+        # option 1: read from file
+        g.read_levels(args.levelfile[0])
+    elif args.levelvalues is not None:
+        # option 2: set values at run time
+        g.assign_levels(args.levelvalues)
+    elif args.generatelevels is not None:
+        # option 3: generate levels
+        check_level_gen(args)
+        minN = min(args.extN)
+        maxN = max(args.extN)
+        g.generate_levels(args.N[0], minN, maxN, mode=args.generatelevels[0])
+    else:
+        raise RuntimeError("Mode for setting level information is not " +
+                           "recognized")
+
+
+def process_tags(tags):
+    """Process the provided tag information to correct format.
+
+    Converts the list of information to a dictionary to be able to pass to the
+    geometry creation step.
+
+    Input:
+    -----
+        tags: list of lists, [[tagname1, tagval1], [tagname2, tagval2], ...]
+
+    Return:
+    -------
+        tagdict: dict, key=TAGNAME, value=TAGVALUE
+    """
+    tagdict = {}
+    for tagset in tags:
+        key = tagset[0]
+        val = float(tagset[1])
+        tagdict[key] = val
+
+    return tagdict
+
+
+def main():
+
+    # get args
+    args = parse_arguments()
+
+    # create instance
+    g = v.IsoVolume()
+
+    # get level info regardless of mode
+    get_levels(args, g)
+
+    # get database information
+    db = os.getcwd() + '/' + args.db[0]
+
+    # run steps depending on mode
+    mode = args.which
+    if mode == 'full' or mode == 'visit':
+        # generate isosurfaces
+        g.generate_volumes(args.meshfile[0], args.dataname[0], db)
+
+    if mode == 'full' or mode == 'moab':
```

Comment:
with previous comment:
```suggestion
    if mode in moab_modes:
```